src/PlatformSockets.c

```diff
@@ -249,6 +262,11 @@ SOCKET bindUdpSocket(int addressFamily, struct sockaddr_storage* localAddr, SOCK
 #endif
     }
 
+#ifdef __3DS__
+    // binding to wildcard port is broken on the 3DS, so we need to define a port manually
+    struct sockaddr_in *n3ds_addr = &bindAddr;
+    n3ds_addr->sin_port = n3ds_udp_port++;
```

Comment:
`sin_port` is expected to be in network byte order, so you should use `htons()` here.

Reply:
Done!